src/logic/BrazeCards.test.ts

```diff
@@ -0,0 +1,247 @@
+                        const result = brazeCards.getCardsForProfileBadge();
+                        expect(result[0].extras).toEqual(extras);
+                    }),
+                );
+            });
+
+            test('returns no cards if the cached feed contains only unsupported cards', () => {
+                // cards with no extras or missing the slotName, so they cannot be handled by this library
+                jest.spyOn(appboy, 'getCachedContentCards').mockImplementation(() => {
+                    return new appboy.ContentCards(
+                        [
+                            // a card for an unsupported slot
+                            testCard('AnotherSlotName', 'card-id-another-slot', {}),
+                            // a card that does not contain a slotname
+                            new appboy.ControlCard(
+                                'card-id-no-slotname',
+                                false,
+                                new Date(),
+                                new Date(new Date().getTime() + 86400),
+                                { extraField: 'field-value' },
+                                false,
+                            ),
+                        ],
+                        new Date(),
+                    );
+                });
+                const brazeCards = new BrazeCards(appboy, errorHandler);
+                const result = brazeCards.getCardsForProfileBadge();
+                expect(result).toEqual([]);
+            });
+        });
+
+        test("The last updated date comes from the appboy instance's last updated date", () => {
+            fc.assert(
+                fc.property(fc.date(), (date) => {
+                    jest.spyOn(appboy, 'getCachedContentCards').mockImplementation(() => {
+                        return new appboy.ContentCards([], date);
+                    });
+                    const brazeCards = new BrazeCards(appboy, errorHandler);
+                    expect(brazeCards.lastUpdated).toEqual(date);
+                }),
+            );
+        });
+    });
+
+    describe('BrazeCard', () => {
+        test('logImpression calls the appboy sdk method', () => {
+            const mockAppboyFn = jest.fn();
+            const mockErrorHandler = jest.fn();
+            const appboyCard = testCard('ProfileBadge', 'card-id', {});
+            jest.spyOn(appboy, 'logCardImpressions').mockImplementation((cards) => {
+                mockAppboyFn(cards);
+                return true;
+            });
+            const brazeCard = new BrazeCard(
+                'card-id',
+                'ProfileBadge',
+                appboyCard,
+                appboy,
+                mockErrorHandler,
+            );
+
+            brazeCard.logImpression();
+
+            expect(mockAppboyFn).toHaveBeenCalledWith([appboyCard]);
+            expect(mockErrorHandler).toHaveBeenCalledTimes(0);
```

Comment:
Super nitpicky, but I think you could also write this as:

```
expect(mockErrorHandler).not.toHaveBeenCalled();
```

which I think reads _marginally_ nicer.